baracoda/formats/generic.py

```diff
@@ -0,0 +1,28 @@
+from baracoda.formats.interfaces import FormatterInterface
+from baracoda.exceptions import UnsupportedEncodingForPrefix
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class GenericBarcodeFormatter(FormatterInterface):
+    def __init__(self, prefix: str):
+        logger.debug(f"Instantiate formatter with {prefix}")
+        if not prefix.isascii():
+            raise UnsupportedEncodingForPrefix("The prefix provided {prefix} is not ASCII")
+
+        self.prefix = prefix
+
+    def barcode(self, value: int) -> str:
+        """
+        Method which returns a barcode with a prefix.
+        If the barcode needs to be converted it is formatted otherwise it is returned as is
```

Comment:
Where is there formatting taking place? In this method?

Reply:
Ah can see not formatting is taking place for this generic formatted, maybe the comment could be updated?